packages/realm-server/worker-manager.ts

```diff
@@ -75,63 +81,93 @@ let isExiting = false;
 process.on('SIGINT', () => (isExiting = true));
 process.on('SIGTERM', () => (isExiting = true));
 
-if (port != null) {
-  // in tests we start a simple TCP server to communicate to the realm when
-  // the worker is ready to start processing jobs
-  let server = createServer((socket) => {
-    log.info(`realm connected to worker manager`);
-    socket.on('data', (data) => {
-      if (data.toString() === 'ready?') {
-        socket.write(isReady ? 'ready' : 'not-ready');
-      }
-    });
-    socket.on('close', (hadError) => {
-      log.info(`realm has disconnected${hadError ? ' due to an error' : ''}.`);
-    });
-    socket.on('error', (err: any) => {
-      console.error(`realm disconnected from worker manager: ${err.message}`);
-    });
-  });
-  server.unref();
+let dbAdapter = new PgAdapter({});
 
-  server.listen(port, () => {
-    log.info(`worker manager listening for realm on port ${port}`);
-  });
+let webServer = new Koa<Koa.DefaultState, Koa.Context>();
+let router = new Router();
+router.head('/', livenessCheck);
+router.get('/', async (ctxt: Koa.Context, _next: Koa.Next) => {
+  let result = {
+    ready: isReady,
+  } as Record<string, boolean | number>;
+  if (isReady) {
+    let [{ queue_depth }] = (await query(dbAdapter, [
+      `SELECT COUNT(*) as queue_depth FROM jobs WHERE status='unfulfilled'`,
+    ])) as {
+      queue_depth: string;
+    }[];
+    result = {
```

Comment:
another metric that might be nice to report on would be uncompleted jobs in jobs_reservations that have been running longer than some threshold and are not expired